core/opendaq/server/include/opendaq/server.h

```diff
@@ -43,6 +48,18 @@ DECLARE_OPENDAQ_INTERFACE(IServer, IBaseObject)
     * @brief Stops the server. This is called when we remove the server from the Instance or Instance is closing.
     */
     virtual ErrCode INTERFACE_FUNC stop() = 0;
+
+    /*!
+    * @brief Gets the server id which will be the key for the server in the module manager.
+    * @param[out] serverId The unique server id.
+    */
+    virtual ErrCode INTERFACE_FUNC getServerId(IString** serverId) = 0;
+
+    /*!
+    * @brief Gets the server configuration properties
+    * @param[out] config The server configuration properties.
+    */
+    virtual ErrCode INTERFACE_FUNC getServerConfig(IPropertyObject** config) = 0;
```

Comment:
These two are available in the IServerType object.

Reply:
we can create server with overridden server with custom properties values  in config. if we move  getServerConfig to IServerType it means that we will have an access not to the actual server config, but to the default one. At that case mdns service would be configured improperly. 

getServerId returns the unique key based on server config (serviceName_cap_port) so it also needs an access to actual config, not the default one